go-controller/pkg/ovn/master.go

```diff
@@ -1481,7 +1488,9 @@ func (oc *Controller) syncNodesRetriable(nodes []interface{}) error {
 			continue
 		}
 
-		oc.deleteNode(nodeSwitch.Name, subnets)
+		if err := oc.deleteNode(nodeSwitch.Name, subnets); err != nil {
+			klog.Warningf("Failed to delete node:%s, err:%v", nodeSwitch.Name, err)
```

Comment:
if we fail to delete a node shouldn't we return an error here?

Reply:
this one was in syncNodes() path so I wasn't sure if I wanted to return error here

Reply:
yeah we will retry sync nodes if it fails. I think this should be an error. wdyt @flavio-fernandes ?

Reply:
I agree. We should ensure all these calls are idem-potent, tho:
```
func (oc *Controller) deleteNode(nodeName string, hostSubnets []*net.IPNet) {
	// Clean up as much as we can but don't hard error
	for _, hostSubnet := range hostSubnets {
		if err := oc.deleteNodeHostSubnet(nodeName, hostSubnet); err != nil {
			klog.Errorf("Error deleting node %s HostSubnet %v: %v", nodeName, hostSubnet, err)
		} else {
			util.UpdateUsedHostSubnetsCount(hostSubnet, &oc.v4HostSubnetsUsed, &oc.v6HostSubnetsUsed, false)
		}
	}
	// update metrics
	metrics.RecordSubnetUsage(oc.v4HostSubnetsUsed, oc.v6HostSubnetsUsed)

	if err := oc.deleteNodeLogicalNetwork(nodeName); err != nil {
		klog.Errorf("Error deleting node %s logical network: %v", nodeName, err)
	}

	if err := oc.gatewayCleanup(nodeName); err != nil {
		klog.Errorf("Failed to clean up node %s gateway: (%v)", nodeName, err)
	}

	if err := oc.joinSwIPManager.ReleaseJoinLRPIPs(nodeName); err != nil {
		klog.Errorf("Failed to clean up GR LRP IPs for node %s: %v", nodeName, err)
	}

	if err := libovsdbops.DeleteNodeChassis(oc.sbClient, nodeName); err != nil {
		klog.Errorf("Failed to remove the chassis associated with node %s in the OVN SB Chassis table: %v", nodeName, err)
	}
}
```

Reply:
@msherif1234 I think you need to turn the warning into the error message, and like Flavio said, ensure the things called in delete node are idempotent

Reply:
done